### PR TITLE
Add aws_iot_certificate resource 

### PR DESCRIPTION
**Pull request by AlexMabry**
Add IoT Certificate resource.  Code is based on work from @jhedev.  This one is super simple.

### CODE REVIEW

**Comment by grubernaut** [comment]
Needs `go fmt` ran on source files to fix failing tests

File: aws/resource_aws_iot_certificate.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	}
+	log.Printf("[DEBUG] Created certificate from csr")
+
+	d.SetId(*out.CertificateId)
+
+	return resourceAwsIotCertificateRead(d, meta)
+}
+
+func resourceAwsIotCertificateRead(d *schema.ResourceData, meta interface{}) error {
+
```
Comment (grubernaut) [inline aws/resource_aws_iot_certificate.go]: Extra line after function header

File: aws/resource_aws_iot_certificate.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	out, err := conn.DescribeCertificate(&iot.DescribeCertificateInput{
+		CertificateId: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		log.Printf("[ERROR] %s", err)
+		return err
+	}
+
+	d.Set("arn", out.CertificateDescription.CertificateArn)
```
Comment (grubernaut) [inline aws/resource_aws_iot_certificate.go]: is `active` also returned from the `DescribeCertificate` API call? If so, we need to set that here as well, to be able to catch any diffs between infra+state.

File: aws/resource_aws_iot_certificate.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		if err != nil {
+			log.Printf("[ERROR] %s", err)
+			return nil
+		}
+
+		// If everything worked, make the old one inactive
+		_, err = conn.UpdateCertificate(&iot.UpdateCertificateInput{
+			CertificateId: aws.String(d.Id()),
+			NewStatus:     aws.String("INACTIVE"),
+		})
```
Comment (grubernaut) [inline aws/resource_aws_iot_certificate.go]: If I'm reading this correctly, if a user updates the `csr`, Terraform will create a new certificate resource, and leak the existing one? Should we delete the original certificate here as well?

File: aws/resource_aws_iot_certificate.go (hunk 2 above)
Comment (AlexMabry) [inline aws/resource_aws_iot_certificate.go]: ok

File: aws/resource_aws_iot_certificate.go (hunk 1 above)
Comment (AlexMabry) [inline aws/resource_aws_iot_certificate.go]: ok

File: aws/resource_aws_iot_certificate.go (hunk 3 above)
Comment (AlexMabry) [inline aws/resource_aws_iot_certificate.go]: There is really no way to update the csr in API.  Would it be more straightforward to delete the old one or just not allow changing it in the first place?

**Comment by AlexMabry** [comment]
I changed it to prevent changing the CSR, since it requires creating a new cert

**Pull request merged**